### PR TITLE
Move kubevirtci k8s startup and dev guide to parent dir

### PR DESCRIPTION
**Pull request by dhiller**
Move the files that are largely the same for each provider into parent
directory, generalize them over the k8s version. Fix some spelling.

Also add an overview of what is automated regarding k8s provider updates
and creation.

Signed-off-by: Daniel Hiller <dhiller@redhat.com>

/cc @fgimenez

### CODE REVIEW

File: cluster-up/cluster/K8S_DEV_GUIDE.md [hunk 1]
```diff
@@ -140,9 +142,13 @@ custom_manifest="/tmp/custom_manifest.yaml"
...
 ```
 * Run ./cluster-provision/k8s/1.21.0/provision.sh, it will create a new provision and test it.
-* Run ./cluster-provision/k8s/1.21.0/publish.sh, it will publish the new created image to docker.io
-* Update k8s-1.21.0 image line at cluster-up/cluster/images.sh, to point on the newly published image.
+
+# Manual steps for publishing a new provider
+
+The steps to create, test and integrate a new KubeVirtCI provider are [mostly automated](./K8S_AUTOMATION.md), but just in case you need to do it manually:
+
+* Run `./cluster-provision/k8s/1.21.0/publish.sh`, it will publish the new created image to quay.io
```
Comment (oshoval) [inline cluster-up/cluster/K8S_DEV_GUIDE.md:150]: maybe parameterize 1.21.0 ?
(its 1.21 now btw)

**Review by oshoval — COMMENTED** [review]
Thanks

File: cluster-up/cluster/K8S_DEV_GUIDE.md [hunk 2]
```diff
@@ -27,9 +35,9 @@ Running `make cluster-up` will deploy a pre-provisioned cluster.
 Upon finishing deployment of a K8s deploy, we will have 3 containers:
 * k8s-1.21 vm container - a container that runs a qemu VM, which is the K8s node, in which the pods will run.
```
Comment (fgimenez) [inline cluster-up/cluster/K8S_DEV_GUIDE.md:36]:
```suggestion
* k8s-${KUBEVIRT_PROVIDER_VERSION} vm container - a container that runs a qemu VM, which is the K8s node, in which the pods will run.
```

File: cluster-up/cluster/K8S_DEV_GUIDE.md [hunk 3]
```diff
@@ -93,7 +101,7 @@ root         1     0 36 13:39 ?        00:05:22 qemu-system-x86_64 -enable-kvm -
 ```
 
 # Flow of K8s provisioning (1.21 for example)
```
Comment (fgimenez) [inline cluster-up/cluster/K8S_DEV_GUIDE.md:103]:
```suggestion
# Flow of K8s provisioning ${KUBEVIRT_PROVIDER_VERSION}
```

File: cluster-up/cluster/K8S_DEV_GUIDE.md [hunk 4]
```diff
@@ -112,11 +120,11 @@ root         1     0 36 13:39 ?        00:05:22 qemu-system-x86_64 -enable-kvm -
 # Flow of K8s cluster-up (1.21 for example)
```
Comment (fgimenez) [inline cluster-up/cluster/K8S_DEV_GUIDE.md:120]:
```suggestion
# Flow of K8s cluster-up ${KUBEVIRT_PROVIDER_VERSION}
```

**Review by fgimenez — COMMENTED** [review]
A few remaining 1.21 references, other than that looks good

**Review by fgimenez — APPROVED** [review]
/approve

Thanks!

**Pull request merged**